confmap/resolver.go

```diff
@@ -171,7 +172,13 @@ func (mr *Resolver) Resolve(ctx context.Context) (*Conf, error) {
 		if err != nil {
 			return nil, err
 		}
-		cfgMap[k] = val
+
+		if v, ok := val.(string); ok && featuregates.UseUnifiedEnvVarExpansionRules.IsEnabled() {
```

Comment:
I think `Test_EscapedEnvVars` only validates output with the feature gate enabled, what's the output if it is disabled?

Reply:
I'll add a test for that as well.

Reply:
Actually it should be handled by this existing test: https://github.com/open-telemetry/opentelemetry-collector/blob/8de544b176179a21637baf094b833969786f4479/confmap/converter/expandconverter/expand_test.go#L116C29-L116C35.  In order to write a test that mimics the functionality with the feature gate disabled I'd have to reproduced the expand converters logic.

Reply:
>In order to write a test that mimics the functionality with the feature gate disabled I'd have to reproduced the expand converters logic in confmap tests.

Thanks for digging into it. That assumes that the expand converter is used, though, right? I guess without this, it should just be that all `$$` remain as `$$` instead of being replaced.

Reply:
Ya without the expand converter in place, there was no escaping.